Homework2/src/problem2/Basket.java

```diff
@@ -0,0 +1,37 @@
+package problem2;
+
+
+import java.util.ArrayList;
+
+public class Basket {
+    private double totalPrice = 0;
+    private double price = 0;
+    private Customer customer;
+    private ArrayList<Product> products = new ArrayList<>();
+
+    public Basket(Customer customer){
+        this.customer = customer;
```

Comment:
si aici ar merge o validare pt a  impiedica crearea unui cos invalid, care e creat cu un customer null, ca sa impiedici exceptiile mai tarziu, cand in getTotalPrice invoci customer.getMembership care iti poate arunca un NullPointerException: astfel arunci de la crearea obiectului o exceptie